ground/src/main/java/com/google/android/ground/persistence/local/room/dao/SubmissionMutationDao.kt

```diff
@@ -37,6 +38,17 @@ interface SubmissionMutationDao : BaseDao<SubmissionMutationEntity> {
     vararg allowedStates: MutationEntitySyncStatus
   ): List<SubmissionMutationEntity>
 
+  @Query(
+    "SELECT COUNT(*) FROM submission_mutation " +
+      "WHERE location_of_interest_id = :locationOfInterestId AND state IN (:allowedStates) " +
+      "AND type = :mutationType"
+  )
+  suspend fun getPendingSubmissionCountByLocationOfInterestId(
```

Comment:
Nit: Technically this can get mutations in any state, not just PENDING. Also, internally we've started abbreviating LOIs, feel free to do the same if you like. Alternatively, since there are actually multiple criteria, I think it's ok to omit that part:

   `getSubmissionMutationCount(...)`